### PR TITLE
Availability Updates - Day, Week & Reorg

### PR DESCRIPTION
**Pull request by sjmcculloch**
- Availability **Day View** implementation
-- View/Edit time segments per day
-- Navigate between days of a request

image

- Availability **Week View** restructure 

image

- New time range modal (to be extended with mobile time segments in future)

image

- Reorganised listing pages with filters

image

- Reorganised edit pages to remove steps

image

### CODE REVIEW

File: web/src/screens/private/Schedules/ViewSchedule.js [hunk 1]
```diff
@@ -29,73 +23,114 @@ import {
...
-    endTimeRange: moment().isoWeekday(1).startOf('week').add(7, 'days')
+    startTimeRange: moment()
+      .isoWeekday(1)
+      .startOf('week')
+      .add(1, 'days')
+      .unix(),
+    endTimeRange: moment()
+      .isoWeekday(1)
+      .startOf('week')
+      .add(8, 'days')
```
Comment (OSPFNeighbour) [inline web/src/screens/private/Schedules/ViewSchedule.js]: add 8 days? might want to comment that to explain why. If i was bug hunting my brain picks this out as an anomaly

File: web/src/screens/private/Schedules/ViewScheduleDay.js [hunk 2]
```diff
@@ -7,32 +7,218 @@ import moment from 'moment';
...
+  onNextDateRange = () => {
+    const currentDay = moment.unix(parseInt(this.props.match.params.time, 10));
+    currentDay.add(1, 'days');
+    const { history } = this.props;
+    history.push(`/schedules/${this.props.match.params.id}/${currentDay.unix()}`);
+  };
+
+  onPreviousDateRange = () => {
+    const currentDay = moment.unix(parseInt(this.props.match.params.time, 10));
+    currentDay.add(-1, 'days');
```
Comment (OSPFNeighbour) [inline web/src/screens/private/Schedules/ViewScheduleDay.js]: isnt that called moment().subtract ?

File: web/src/screens/private/Schedules/ViewSchedule.js (hunk 1 above)
Comment (sjmcculloch) [reply web/src/screens/private/Schedules/ViewSchedule.js]: First ISO day starts on Sunday - so startTimeRange is + 1 day. I'll move this to a selector.

File: web/src/screens/private/Schedules/ViewScheduleDay.js (hunk 2 above)
Comment (sjmcculloch) [reply web/src/screens/private/Schedules/ViewScheduleDay.js]: Same thing - it's how it's done in .NET using add I think (old habit)

**Pull request merged**